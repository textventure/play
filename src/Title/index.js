import React, { Component } from 'react';
import Typography from '@material-ui/core/Typography';
import Card from '../Card';
import Choice from '../Choice';
import { formatAuthor, getKey, getValue } from '../helpers/util';

export default class Title extends Component {
  static defaultProps = {
    classes: {},
    config: {},
  };

  render() {
    const { classes, config, displayAction, selectChoice } = this.props;
    let { title, author, description, start } = config;
    author = formatAuthor(config.author);

    return (
      <Card>
        <div className={classes.cardContent}>
          {/* title */}
          {title && (
            <Typography component="h1" gutterBottom variant="headline">
              {title}
            </Typography>
          )}

          {/* author */}
          {author && (
            <Typography variant="caption" gutterBottom>
              By {author}
            </Typography>
          )}

          {/* description */}
          {description && <Typography gutterBottom>{description}</Typography>}
        </div>

        {/* action */}
        {displayAction && (
          <Choice
            choiceId={getValue(start)}
            className={classes.button}
            selectChoice={selectChoice}
          >
            {getKey(start)}
          </Choice>
        )}
      </Card>
    );
  }
}
