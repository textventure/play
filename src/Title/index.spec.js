import React from 'react';
import { shallow } from 'enzyme';
import Title from '.';

let wrapper;

describe('without props', () => {
  it('renders without error', () => {
    expect(() => (wrapper = shallow(<Title />))).not.toThrow();
  });

  it('renders correctly', () => {
    expect(wrapper.getElement()).toMatchSnapshot();
  });
});

describe('with props', () => {
  const props = {
    classes: {
      button: 'button',
      cardContent: 'cardContent',
    },
    config: {
      author: 'Author',
      description: 'Description',
      title: 'Title',
      start: { Start: 'start' },
    },
    displayAction: true,
    selectChoice: Function,
  };

  beforeAll(() => {
    wrapper = shallow(<Title {...props} />);
  });

  it('renders correctly', () => {
    expect(wrapper.getElement()).toMatchSnapshot();
  });

  it('passes props.selectChoice to <Choice>', () => {
    expect(wrapper.find('WithStyles(Choice)').prop('selectChoice')).toBe(
      props.selectChoice
    );
  });
});
