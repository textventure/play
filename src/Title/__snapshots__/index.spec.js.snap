// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with props renders correctly 1`] = `
<WithStyles(Card)>
  <div
    className="cardContent"
  >
    <WithStyles(Typography)
      component="h1"
      gutterBottom={true}
      variant="headline"
    >
      Title
    </WithStyles(Typography)>
    <WithStyles(Typography)
      gutterBottom={true}
      variant="caption"
    >
      By 
      Author
    </WithStyles(Typography)>
    <WithStyles(Typography)
      gutterBottom={true}
    >
      Description
    </WithStyles(Typography)>
  </div>
  <Choice
    choiceId="start"
    className="button"
    selectChoice={[Function]}
  >
    Start
  </Choice>
</WithStyles(Card)>
`;

exports[`without props renders correctly 1`] = `
<WithStyles(Card)>
  <div
    className={undefined}
  >
    
  </div>
</WithStyles(Card)>
`;
